tests/test_basis_set.py

```diff
@@ -419,3 +422,442 @@ def test_quick_step() -> None:
         ],
     }
     # TODO: generate a QuickStep generator in the CP2K plugin
+
+
+# -------------------------------
+# AtomCenteredFunction (combined shells, padding/truncation, inference)
+# -------------------------------
+
+
+def test_acf_infers_n_primitive_and_ell() -> None:
```

Comment:
`pytest.mark.parametrize` for testing various edge cases, e.g. `n_primitive == 0` (when the parser doesn't populate the section), etc.

Reply:
The same comment may apply to some of the other tests below. I leave it up to your discretion which you extend.